docs/Using Fleet/Scripts.md

```diff
@@ -0,0 +1,74 @@
+# Scripts
+
+_Available in Fleet Premium_
+
+In Fleet you can execute a custom script to remediate an issue on your macOS, Windows, and Linux hosts.
+
+Shell scripts are supported on macOS and Linux. All scripts will run in the host's (root) default shell (`/bin/sh`). Other interpreters are not supported yet.
+
+PowerShell scripts are supported on Windows. Other types or scripts are not supported yet.
+
+Script execution is disabled by default. Continue reading to learn how to enable scripts.
+
+## Execute a script
+
+You can execute a script using the `fleetctl` command-line interface.
+
+To execute a script, we will do the following steps:
+1. Enable script execution
+2. Write a script
+3. Run the script
+4. View script activities in the UI
+
+### Step 1: enable script execution
+
+If you use Fleet's macOS MDM features, scripts are automatically enabled for macOS hosts that have MDM turned on. You're set!
+
+If you don't use MDM features, to enable scripts, we'll deploy a fleetd agent with scripts enabled:
+
+1. Generate a new fleetd agent for macOS, Windows, or Linux using the `fleetctl package` command with the `--script-execution` flag. 
+
+2. Deploy fleetd to your hosts. If your hosts already have fleetd installed, you can deploy the new fleetd on-top of the old installation.
+
+Learn more about generating a fleetd agent and deploying it [here](./enroll-hosts.md#enroll-hosts-with-fleetd).
+
+### Step 2: write a script
+
+As an example, we'll write a shell script for a macOS host that downloads a Fleet wallpaper and set the host's wallpaper to it.
+
+To run the script, we'll need to create a `set-wallpaper-to-fleet.sh` file locally and copy and paste this script into this `.sh` file:
+
+```sh
+wallpaper="/tmp/wallpaper.png" 
+
+curl --fail https://fleetdm.com/images/wallpaper-cloud-city-1920x1080.png -o $wallpaper
+
+osascript -e 'tell application "Finder" to set desktop picture to POSIX file "'"$wallpaper"'"' 
+```
+
+### Step 3: run the script
+
+1. Run the `fleetctl run-script --script_path=set-wallpaper-to-fleet.sh --host=hostname` command.
+
+> Replace --host flag with your target host's hostname respectively.
+
+2. Look at the on-screen information. In the output you'll see the script's exit code and output.
+
+### Step 4: view script activities in the UI
```

Comment:
I already left some comments below, but after thinking about it some more: I suggest cutting step 4 altogether and adding a note at the end of "Run the script" explaining that this generates an activity feed entry, linking [here](https://fleetdm.com/docs/using-fleet/audit-logs#type-code-ran-script-code) for details. (I think everything in this section is already documented elsewhere, and less duplication means a smaller chance of docs getting out-of-date.)

Reply:
I moved step 4 into 2 sentences at the bottom of step 3:

![Screenshot 2023-09-11 at 5 51 52 PM](https://github.com/fleetdm/fleet/assets/47070608/208564ed-dfcc-4b2f-84d0-c358208aec2c)

I don't think we point the user to the Activity feed UI on the Audit logs page.

Reply:
> I don't think we point the user to the Activity feed UI on the Audit logs page.

On second thought, maybe we should just add this. To avoid duplication in the future.

Reply:
Editing the Audit logs page is a little tricky. I can do it but won't get to it before tomorrow. Maybe to unblock this PR and fix the [broken link](fleetdm.com/using-fleet/scripts) we can merge it as is? 

I took a note to make this change tomorrow.